File: 1359. Count All Valid Pickup and Delivery Options/count-all-valid-pickup-and-delivery-options.rs
// 1359. Count All Valid Pickup and Delivery Options (2/7/54148)
// Runtime: 1 ms (0.00%) Memory: 2.06 MB (0.00%) 

impl Solution {
    pub fn count_orders(n: i32) -> i32 {
        let mut ans: i64 = 1;
        let modulo: i64 = 1000000007;
    
        for i in 1i64..=(n as i64) {
            ans *= 2 * i - 1;
            ans %= modulo;
        }

        for i in 1i64..=(n as i64) {
            ans *= i;
            ans %= modulo;
        }

        return ans as i32;
    }
}

File: 1359. Count All Valid Pickup and Delivery Options/count-all-valid-pickup-and-delivery-options.java
// 1359. Count All Valid Pickup and Delivery Options (2/9/54148)
// Runtime: 0 ms (94.99%) Memory: 41.28 MB (35.76%) 

class Solution {
    public int countOrders(int n) {
        long ans = 1;
        long mod = 1000000007;
        
        for (int i = 2; i <= n; i++) {
            ans *= 2 * i - 1;
            ans %= mod;
        }
        
        for (int i = 2; i <= n; i++) {
            ans *= i;
            ans %= mod;
        }
        
        return (int)ans;
    }
}